cmplxfoil/CMPLXFOIL.py

```diff
@@ -0,0 +1,1120 @@
+            failCheck = {}
+            self.checkSolutionFailure(aeroProblem, failCheck)
+            cl = float(self.funcs[aeroProblem.name]["cl"])
+            print(f"Alpha: {aeroProblem.alpha:>6.3f}, CL: {cl:>7.6f}")
+            res = cl - CLStar
+
+            # Check for convergence and divergence
+            hasConverged = abs(res) < tol
+            if hasConverged:
+                print("Converged!")
+                break
+            elif abs(res > maxRes):
+                break
+
+            # If not converged, compute the next alpha
+            if useNewton:
+                aeroProblem.alpha += 1e-200 * 1j
+                self.__call__(aeroProblem, useComplex=True, deriv=True)
+                dCLdAlpha = np.imag(self.funcsComplex[aeroProblem.name]["cl"]) * 1e200
+                aeroProblem.alpha = np.real(aeroProblem.alpha)
+            else:
+                # If using secant, we can only compute dCLdAlpha from the second iteration onwards
+                if resPrev is not None:
+                    dCLdAlpha = (res - resPrev) / (aeroProblem.alpha - alphaPrev)
+            resPrev = res
+            alphaPrev = aeroProblem.alpha
+
+            # Update the alpha either using dCLdAlpha or delta, or backtracking if something went wrong
+            if dCLdAlpha == 0.0 or failCheck["fail"]:
+                aeroProblem.alpha *= 0.9
+            elif dCLdAlpha is not None:
+                aeroProblem.alpha = np.clip(aeroProblem.alpha - res / dCLdAlpha, alphaBound[0], alphaBound[1])
+            else:
+                aeroProblem.alpha = aeroProblem.alpha + delta
+        if not hasConverged:
+            print("Did not converge :(")
+        return hasConverged
+
+    def writeSolution(self, outputDir=None, baseName=None, number=None):
+        """This is a generic shell function that potentially writes the various output files. The intent is that the
+        user or calling program can call this file and CMPLXFOIL write all the files that the user has defined. It is
+        recommended that this function is used along with the associated logical flags in  the options to determine the
+        desired writing procedure.
+
+        Parameters
+        ----------
+        outputDir : str
+            Use the supplied output directory
+        baseName : str
+            Use this supplied string for the base filename. Typically only used from an external solver.
+        number : int
+            Use the user supplied number to index solution. Again, only typically used from an external solver.
+        """
+        if outputDir is None:
+            outputDir = self.getOption("outputDirectory")
+
+        if baseName is None:
+            baseName = self.curAP.name
+
+        # If we are numbering solution, it saving the sequence of
+        # calls, add the call number
+        if number is not None:
+            # We need number based on the provided number:
+            baseName = baseName + "_%3.3d" % number
```

Comment:
It would require many, many iterations, but if I think this could get truncated. Would it be worth leaving this "unformatted" so that the number doesn't risk getting truncated in an edge case (i.e. using as many digits as it wants)?

Reply:
After in person discussion, we decided to leave it here since ADflow uses the same. It's also a minimum so it won't truncate (thanks Sabet)